iocore/net/SSLUtils.cc

```diff
@@ -1163,7 +1171,7 @@ setClientCertLevel(SSL *ssl, uint8_t certLevel)
   }
 
   Debug("ssl", "setting cert level to %d", server_verify_client);
-  SSL_set_verify(ssl, server_verify_client, nullptr);
+  SSL_set_verify(ssl, server_verify_client, ssl_verify_client_callback);
```

Comment:
If the set_verify could be done earlier with SSL_CTX_set_verify(), i.e. globally for this ctx, would that eliminate the need to set it again for each SSL object?

Reply:
The decision of whether to require a certificate from the user agent is not made globally.  The ssl_server_name logic is executed after the SSL object has already been created, and this logic can change our desired stance on requiring a client certificate or not.